### PR TITLE
Add compareTo to Timestamp.

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: api/src/main/java/io/opencensus/common/Timestamp.java [hunk 1]
```diff
@@ -130,6 +130,26 @@ public Duration subtractTimestamp(Timestamp timestamp) {
     return Duration.create(durationSeconds, durationNanos);
   }
 
+  /**
+   * Compares this {@code Timestamp} to the specified {@code Timestamp}.
+   *
+   * @param otherTimestamp the other {@code Timestamp} to compare to, not null.
```
Comment (g-easy) [inline api/src/main/java/io/opencensus/common/Timestamp.java]: AFAICT, @code doesn't work in @param lines.

File: api/src/main/java/io/opencensus/common/Timestamp.java [hunk 2]
```diff
@@ -130,6 +130,26 @@ public Duration subtractTimestamp(Timestamp timestamp) {
...
+   * @return the comparator value: zero if equal, negative if this timestamp happens before otherTimestamp, positive if after.
+   * @throws NullPointerException if otherTimestamp is null.
+   */
+  @Override
+  public int compareTo(Timestamp otherTimestamp) {
+    int cmp = compareLong(getSeconds(), otherTimestamp.getSeconds());
+    if (cmp != 0) {
+      return cmp;
+    }
+    return getNanos() - otherTimestamp.getNanos();
```
Comment (g-easy) [inline api/src/main/java/io/opencensus/common/Timestamp.java]: Should this be compareLong() also? So the output is always {-1, 0, +1}?

File: api/src/test/java/io/opencensus/common/TimestampTest.java [hunk 3]
```diff
@@ -135,7 +135,19 @@ public void timestampSubtractTimestamp_NegativeResult() {
   }
 
   @Test
-  public void testTimestampEqual() {
+  public void timestamp_CompareTo() {
+    assertThat(Timestamp.create(0, 0).compareTo(Timestamp.create(0, 0))).isEqualTo(0);
+    assertThat(Timestamp.create(24, 42).compareTo(Timestamp.create(24, 42))).isEqualTo(0);
+    assertThat(Timestamp.create(-24, 42).compareTo(Timestamp.create(-24, 42))).isEqualTo(0);
```
Comment (g-easy) [inline api/src/test/java/io/opencensus/common/TimestampTest.java]: Could you please add a test for negative time without equality?
e.g. T(-24,1) happens before T(-24,2) ?

Comment (bogdandrutu) [inline api/src/test/java/io/opencensus/common/TimestampTest.java]: Done.

**Comment by bogdandrutu** [comment]
PTAL
